### PR TITLE
fix(schedule): fail to filter by tenantId and clusterId

### PR DESCRIPTION
**Pull request by guowl3**
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by kiko-art — DISMISSED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java [hunk 1]
```diff
@@ -789,8 +805,16 @@ public Page<ScheduleTaskListOverview> listScheduleTaskOverviewByScheduleType(@No
...
-                    authenticationFacade.currentOrganizationId(), params.getTenantId()));
+            List<Long> datasourceIds = connectionService.innerListIdByOrganizationIdAndTenantId(
+                    authenticationFacade.currentOrganizationId(), params.getTenantId());
+            if (datasourceIds.isEmpty()) {
+                return Page.empty();
+            }
+            if (params.getDataSourceIds().isEmpty()) {
+                params.getDataSourceIds().addAll(datasourceIds);
+            } else {
+                params.getDataSourceIds().retainAll(datasourceIds);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java:816]: why should we retain all?
the familiar logic above does not do this

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/server/TaskMonitor.java [hunk 2]
```diff
@@ -181,6 +181,7 @@ private void doFinal() {
         uploadLogFileToCloudStorage(finalResult);
 
         Map<String, String> latestLogMeta = finalResult.getLogMetadata();
+        log.info("Task id: {}, latest log meta: {}", getJobId(), latestLogMeta);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/server/TaskMonitor.java:184]: will it print a lot of log?

Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/server/TaskMonitor.java:184]: just print it when the task is finished

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java (hunk 1 above)
Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java:816]: when the filter criteria include both tenantId and clusterId, the union of the two needs to be taken.

**Review by kiko-art — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**